### PR TITLE
Fix #169: No compilation error when assigning a i32 value to a []i32 …

### LINKED ISSUE #169: No compilation error when assigning a i32 value to a []i32 variable
CX version 0.5.18
```
package main

func main()(){
    var sa []i32
    sa = 33

    var sb []str
    sb = "33"
}
```

### PR DESCRIPTION
**Pull request by amherag**
…variable.

Fixes #169

Changes:
- Changes to how CX determines the type of arguments
- CX can now reference inner structures like &slice[0] or &struct.field

Does this change need to mentioned in CHANGELOG.md?
Yes.

### CODE REVIEW

File: cx/opcodes_bare.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ var CorePackages = []string{
 // op codes
 // nolint golint
 const (
-	OP_IDENTITY = iota
+	OP_IDENTITY = iota + 1
```
Comment (ingwal) [inline cx/opcodes_bare.go]: Hmm, why this change?  Was there some collision before?

File: cxgo/actions/functions.go [hunk 2]
```diff
@@ -437,10 +437,10 @@ func GetFormattedType(arg *CXArgument) string {
 			// base type
 			if elt.CustomType != nil {
 				// then it's custom type
-				typ = elt.CustomType.Name + typ
+				typ += typ + elt.CustomType.Name
```
Comment (ingwal) [inline cxgo/actions/functions.go]: This looks like a bug to me.  Are you sure it should be 'typ += typ + ...' ?
If so, there should definitely be a comment to explain it.

File: cx/opcodes_bare.go (hunk 1 above)
Comment (amherag) [reply cx/opcodes_bare.go]: Yeah. Some operators are associated to standard library functions and others to user-made functions. If it's a user-made function, the OpCode == 0, but this also happens if it's a call to OP_IDENTITY (assignment). The way to check if it was a standard library function is to check CXFunction.IsNative, but I decided to add an extra check and just add +1 to it. We should get rid of IsNative when we redesign the parser and just check OpCode.

File: cxgo/actions/functions.go (hunk 2 above)
Comment (amherag) [reply cxgo/actions/functions.go]: Ooops. I think in perfect scenarios it would have never been a bug, because when that part is called `typ` should always been empty. But it's still wrong :) I'll fix it.
